### ISSUE TITLE
group dialog suggestions

### ISSUE THREAD

**Issue by raveit65** [zhuyaliang/user-admin#10]
First i wanna say that you did a great job again.
I found some small issues.
1. when i had create a new group, the new group isn't visible in switch-groups tab.
I need to close/reopen the group-dialow window to see the new group in switch-groups tab.

2. see
![group-admin-dialog](https://user-images.githubusercontent.com/961604/51696837-b5976000-2006-11e9-9677-96d5e35ef6f9.png)
For what is the `return` button good?
Clicking the button gives me a segfault.
```
[rave@mother ~]$ mate-user-admin 
Segmentation fault (core dumped)
```
3. A suggestions/feature request
Can we have a dialog window with normal window decoration?
Or a gsettings key to switch between headerbar and metacity decoration?
You only need to move the unlock button from headerbar.

**Comment by raveit65** [comment]
4. Why not moving group-service to user-admin to avoid packaging an extra package?

**Comment by zhuyaliang** [comment]
@raveit65 
Thank you very much for your testing and advice.
Let me start with the first problem you have encountered.
*  When a new group is created, the new group is added to the tail of the _switch-group tab_ without sorting. So you need to drag the scrollbar to the end to see the new group,I made a change, and when the new group is created successfully, I will prompt you how to view the new group.
* The return button doesn't work. I've removed it.
* I think it would be better to separate them. Group-service provides APIs in the form of dynamic libraries, similar to accouts service,I feel that some projects will need group-service alone。I think it would be better to separate them. Group-service provides APIs in the form of dynamic libraries, similar to accouts service. Group-service can't only serve user-admin, I think other projects will also need it.
* >A suggestions/feature request
Can we have a dialog window with normal window decoration?
Or a gsettings key to switch between headerbar and metacity decoration?
You only need to move the unlock button from headerbar.

I'm sorry I didn't understand you mean.Do you mean to use the dialog window?

**Comment by raveit65** [comment]
> I'm sorry I didn't understand you mean.Do you mean to use the dialog window?

I mean the main group-manage window which use a headerbar.
The headerbar is the place where the unlock button is located.
A normal windows decoration is used by main window of user-admin.

**Comment by raveit65** [comment]
Your changes are working fine.

**Comment by zhuyaliang** [comment]
Thank you for your advice，I didn't reply in time for some special reasons.
>Or a gsettings key to switch between headerbar and metacity decoration?

This way i think it's better.

**Comment by zhuyaliang** [comment]
@raveit65  I'm sorry that it took so long to reply to you,  I made some changes, but I still think the unlock button will be more beautiful in the title bar。If you think it's OK, I'll release new releases.

**Comment by raveit65** [comment]
Now all windows use headerbars. This wasn't my suggestion. MATE default is to use metacity-theme windows decorations.
Now it looks like a gnome application :/

**Comment by raveit65** [comment]
Btw. using headerbars for windows is problematic for MATE, because you can move the window under the top panel with the mouse device.

**Comment by zhuyaliang** [comment]
@raveit65 I now use *g_settings_get* ```org.mate.desktop.interface.gtk-dialogs-use-header``` to determine whether header-bar is used or not.And some display effects have been improved，I hope you can test him and give me some advice。

**Comment by raveit65** [comment]
Amazing, this works very nice and the metacity-theme decoration is back.
You make me very happy :)
I will update German translations the next days.

**Comment by zhuyaliang** [comment]
@raveit65 
Thank you for your test. After you update the translation, I will release a new version.

**Comment by raveit65** [comment]
Do you like to add user-admin to MATE at transifex for translations?
https://www.transifex.com/mate/MATE/dashboard/
I can do that......

**Comment by zhuyaliang** [comment]
@raveit65 
I agree with this idea.I've created users in transifex。In addition, I released version 1.4.0

**Comment by raveit65** [comment]
@zhuyaliang 
Just create a . tx folder in your local repo. An example you can found here (This is branch specific)
https://github.com/mate-desktop/marco/tree/master/.tx
The file could be look like this (without specify a branch):
```
[main]
host = https://www.transifex.com
 
[MATE.user-admin]
file_filter = po/<lang>.po
source_file = po/user-admin.pot
source_lang = en
type = PO
minimum_perc = 2
```
Help docs:
https://docs.transifex.com/getting-started-as-a-localization-manager/1-set-up-your-projects
You need to install the transifex-client first.
If you have set up the.tx folder local than commit this.
Than i can add user-admin to Mate group at transifex.
I will enable auto-update from your resourse file at github https://github.com/zhuyaliang/user-admin/blob/master/po/mate-user-admin.pot
That means, you only need to update and commit your .pot file and transifex will update the resource 2 times a day.
For getting the translations from transifex server after a while do` tx pull -af` .
OK, if you need help ask me and let me know if your config file is ready.

**Comment by raveit65** [comment]
Thanks for releasing 1.4.0. I need to do first a review request for group-admin for adding it to fedora.

**Comment by zhuyaliang** [comment]
@raveit65 I'm very sorry, it took me so long to get back to you because of some other  work. I have the ```.tx/config```  ready.

**Comment by raveit65** [comment]
@zhuyaliang 
It's better to change the `slug`, see my comment at https://github.com/zhuyaliang/user-admin/commit/3371fc713a4669d7832d2bdfb6bec5fd525663a0#commitcomment-33259793
Than i can add user-admin quickly to transifex.

**Comment by zhuyaliang** [comment]
@raveit65 This is a very late reply. I modified ```slug``` as you suggested.

**Comment by raveit65** [comment]
@zhuyaliang 
It's done. https://www.transifex.com/mate/MATE/mate-user-admin/
The resource will be updated automatic every time you commit a change for
https://github.com/zhuyaliang/user-admin/blob/master/po/mate-user-admin.pot
After a while you can pull translations with
`tx pull -af`

**Comment by raveit65** [comment]
Don't forget to update https://github.com/zhuyaliang/user-admin/blob/master/po/LINGUAS
when new languages are added by transifex, after `tx pull -af` ;)

**Comment by kyrios123** [comment]
@raveit65  It looks like the .po file at least for french wasn't imported in transifex because most strings are currently not translated while almost all of them were translated already. Is that normal ?

![image](https://user-images.githubusercontent.com/7815316/57330300-e57df580-7115-11e9-9bcd-fc4e11a7600c.png)

**Comment by raveit65** [comment]
Opps, i fogot to upload .po file, only the .pot file was uploaded.
It is fixed now. French language is at 91,47% :)

**Comment by kyrios123** [comment]
Thanks :)

**Comment by zhuyaliang** [comment]
@raveit65 I updated the user-managed version。```1.5.0``` and  ```1.5.1```

**Comment by raveit65** [comment]
@zhuyaliang 
Thanks for new updates.
But is this really needed to force this groups for new users?
```
#Default Add Groups for New Users
nugroups =mail;audio;video;lightdm
```
I think i will comment this out for fedora.

Btw. Should i add  group-service to transifex?

**Comment by zhuyaliang** [comment]
@raveit65  I see what you mean. The default group of new users should be empty, and the data （groups name）should be filled by the users themselves. But I have released the version. How about solving this problem in the next version?
Thank you very much for your help. I will inform you as soon as I prepare the ```.tx``` file of ```group-service```.

**Comment by raveit65** [comment]
> How about solving this problem in the next version?

Np for me , i can edit the file during RPM building of current release, `grep` is my friend :)
Thanks for doing all that work for this very useful application for MATE desktop.
Fedora users asked me to allow mate-user-admin for xfce ;)
https://bugzilla.redhat.com/show_bug.cgi?id=1732687
https://src.fedoraproject.org/rpms/mate-user-admin/c/09a889d421d07e0382137c4c4ed6fd962dc8550e?branch=master

**Comment by raveit65** [comment]
old report, closing
